extensions/resource-deployment/src/services/platformService.ts

```diff
@@ -76,19 +97,106 @@ export class PlatformService implements IPlatformService {
 		return fs.promises.readFile(filePath, 'utf8');
 	}
 
-	runCommand(command: string, options?: CommandOptions): Promise<string> {
+	public logToOutputChannel(data: string | Buffer, header?: string): void {
+		data.toString().split(/\r?\n/)
+			.forEach(line => {
+				this._outputChannel.appendLine(header ? header + line : line);
+			});
+	}
+
+	private outputDataChunk(data: string | Buffer, outputChannel: vscode.OutputChannel, header: string): void {
+		data.toString().split(/\r?\n/)
+			.forEach(line => {
+				outputChannel.appendLine(header + line);
+			});
+	}
+
+	async runCommand(command: string, options?: CommandOptions, sudo?: boolean, commandTitle?: string): Promise<string> {
+		if (commandTitle !== undefined && commandTitle !== null) {
+			this._outputChannel.appendLine(`\t<<<${commandTitle}>>>`);
+		}
+		console.log(`TCL: PlatformService -> options: ${JSON.stringify(options)}`);
+		if (sudo) {
+			return this.runSudoCommand(command, options, this._outputChannel);
+		} else {
+			return this.runStreamedCommand(command, options, this._outputChannel);
+		}
+	}
+
+	private runSudoCommand(cmd: string, options: CommandOptions | undefined, outputChannel: vscode.OutputChannel): Promise<string> {
```

Comment:
Promise marked as async

Reply:
Can you clarify how this is an issue? If you mark as async, it will compile into an async method which doesn't necessarily make sense / adds overhead. Async should only be used if there is an `await` in the code. In cases where we intentionally create Promises, so long as they return a promise and mark themselves as such we should be fine.

Reply:
This rule from palantir explains a rationale:
https://palantir.github.io/tslint/rules/promise-function-async/

Reply:
I agree with aaomidi, our code base will benefit a lot we marked all/most functions/methods that reurn a Promise as async.

Reply:
actually I am going to do one better and convert the body of these functions where possible or where it makes sense to actually use await and let the Promise thingy happen in the background.

Reply:
By truly converting to async/await pattern I am addressing most of aaomidi's concern but I am not marking any left over funcitions that return promise but do not have await in them with async, until I hear from others. Only once left are fairly generic methods of me converting callback based functions that cannot be promisified by wrapping in util.promisify. So there is no business code in them.